include/dav1d/picture.rs

```diff
@@ -220,6 +224,58 @@ impl Default for Rav1dPicture {
     }
 }
 
+impl Rav1dPicture {
+    #[inline(always)]
+    pub fn data_is_null(&self) -> bool {
+        self.data[0].is_null()
+    }
+
+    // TODO(kkysen) Temporary, but for now guarantees aliased XOR mutable.
+    #[inline(always)]
+    pub unsafe fn data(&self) -> [*const c_void; 3] {
+        self.data.map(|data| data.cast_const())
+    }
+
+    /// # Safety
+    ///
+    /// This must be the only mutable access to the pixel data.
+    ///
+    /// TODO(kkysen) Replace this with a [`Mutex`].
+    #[inline(always)]
+    pub unsafe fn data_mut_unchecked(&self) -> [*mut c_void; 3] {
+        self.data
+    }
+
+    // TODO(kkysen) Temporary, but for now guarantees aliased XOR mutable.
+    #[inline(always)]
+    pub unsafe fn data_mut(&self) -> [*mut c_void; 3] {
+        let ref_count = &(*self.r#ref).ref_cnt;
+        let ref_count = unsafe { mem::transmute::<&i32, &AtomicI32>(ref_count) };
+        assert_eq!(ref_count.load(Ordering::SeqCst), 1);
+        self.data_mut_unchecked()
+    }
+
+    #[inline(always)]
+    pub fn set_data(&mut self, data: [*mut c_void; 3]) {
+        self.data = data;
+    }
+}
+
+pub(crate) unsafe fn rav1d_picture_unref_internal(p: &mut Rav1dPicture) {
+    let Rav1dPicture {
+        m: _,
+        data,
+        mut r#ref,
+        ..
+    } = mem::take(p);
+    if !r#ref.is_null() {
+        if validate_input!(!data[0].is_null()).is_err() {
+            return;
+        }
+        rav1d_ref_dec(&mut r#ref);
+    }
+}
```

Comment:
I moved this to have access to the now private `data` field.  Right now things are split between `include/` and `src/`; eventually we'll want to consolidate the related ones into single `mod`ules, though that's not very important yet.  But since some fields are private now, we need to for some methods.  Though eventually this `fn` will be replaced by an auto-generated `impl Drop`, so it won't matter.

Reply:
Why not make this a method of `Rav1dPicture`?

Reply:
It will be deleted eventually, replaced with just a `mem::take`.

Reply:
ok, makes sense